### DISCUSSION TITLE
Click on file path to open it in your favourite text editor

### DISCUSSION THREAD

**Discussion by luigifab** [OpenMage/magento-lts#3657] [Show & Tell]
Every good OpenMage/Magento1/<del>Magento2</del> developer knows this page: [example](https://luigifab.github.io/webext-openfileeditor/)
This is the error page when an error occurs in developer mode.

But what do you do to open the file in your text-editor/ide ?

Stop wasting your time, install and configure this _HIGHLY EXPERIMENTAL_ [web extension](https://github.com/luigifab/webext-openfileeditor) for Firefox (for now),
then click on the link: and tada the file is opened!

**Comment by elidrissidev** [comment]
I see what you did there ;)

VSCode and PHPStorm both support custom URIs that can be used to open files. We could add a small select box to choose between them and then update the `a` tags with some js code for the selected editor. What do you think?

VSCode: `vscode://file/var/www/openmage/app/Mage.php:743`
PHPStorm: `phpstorm://open?url=file:///var/www/openmage/app/Mage.php&line=743`

**Reply by luigifab** [reply]
I didn't know about these links!
But I'm compatible with them with my customizable bash script.

**Reply by luigifab** [reply]
Added with #3679.

**Reply by pquerner** [reply]
I've used that phpstorm protocol in the past, it even has somewhat support in AoE_TemplateHints. (Called something else there, since before those protocols extisted, you had to use a 3rd party module in PHPStorm)

But since everything is now in docker containers its wierd to get to the file path on the host system. Often you have to bring a ENV variable into the mix and its a little messy.

**Reply by luigifab** [reply]
From https://stackoverflow.com/a/59929015/2980105, with Symfony+Docker, it seems there is a tricks.
Not sure if how it works. Perhaps it's possible to add the same in 3679 from env variable.

I forgot OpenMage Template Hints!
I'm going to update them in same way.
